elements/rh-button-group/rh-button-group.ts

```diff
@@ -0,0 +1,111 @@
+import { LitElement, html } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+import { property } from 'lit/decorators/property.js';
+import { queryAssignedElements } from 'lit/decorators/query-assigned-elements.js';
+
+import { themable } from '@rhds/elements/lib/themable.js';
+import styles from './rh-button-group.css';
+import { Logger } from '@patternfly/pfe-core/controllers/logger.js';
+
+// ✅ Import RhButton type for IntelliSense
+import type { RhButton } from '@rhds/elements/rh-button/rh-button.js';
+
+// ✅ Side-effect imports to register elements for runtime & linter
+import '@rhds/elements/rh-button/rh-button.js';
+import '@rhds/elements/rh-button-group/rh-button-group.js';
+
+@customElement('rh-button-group')
+export class RhButtonGroup extends themable(LitElement) {
+  static readonly styles = [styles];
+  #logger = new Logger(this);
+
+  /**
+   * Role of the group:
+   * - "group": Each button tabbable (normal tab order)
+   * - "toolbar": Roving tabindex (WAI-ARIA Toolbar pattern)
+   */
+  @property({ reflect: true })
+  role: 'group' | 'toolbar' = 'group';
+
+  /**
+   * Slotted buttons: can be native <button> or <rh-button>
+   */
+  @queryAssignedElements({ flatten: true })
+  private _buttons!: (HTMLButtonElement | RhButton)[];
+
+  override firstUpdated() {
+    if (!this._buttons?.length) {
```

Comment:
this will likely fail in SSR, since we don't have access to children in the SSR nodejs context